cirq/iqm/iqm_device_test.py

```diff
@@ -0,0 +1,58 @@
+import cirq
+import pytest
+
+
+def test_valid_operations():
+    adonis = cirq.iqm.Adonis()
+
+    adonis.validate_operation(cirq.GateOperation(cirq.X,
+                                                 [cirq.GridQubit(0, 1)]))
+
+    adonis.validate_operation(
```

Comment:
I would add tests for all 5 supported operations.

Reply:
Wouldn't that just repeat the exact logic of the system-under-test? Not sure about the value of that.

Reply:
The value comes in future-proofing for later improvements.  Let's say the validate_operation logic is changed so that measurement can only happen on adjacent qubits or something like that, but the logic is flawed.  All tests would pass because nothing tests measurement gates.

Reply:
I respectfully disagree, but debating this is probably not going to take us anywhere, so I will add all operations in the test to move forward.

Reply:
All gates added in https://github.com/quantumlib/Cirq/pull/2701/commits/fee75d2ba8e5589bfa00eb8d9cb0d9bbde66075c

Reply:
Measurement operation added in https://github.com/quantumlib/Cirq/pull/2701/commits/e86ab87a594d46a1da428c9cfdc9c4c586073e25